spyder/widgets/ipythonconsole/client.py

```diff
@@ -203,6 +204,10 @@ def configure_shellwidget(self, give_focus=True):
         self.shellwidget.sig_show_syspath.connect(self.show_syspath)
         self.shellwidget.sig_show_env.connect(self.show_env)
 
+        #To sync global working directory
+        self.shellwidget.executing.connect(self.shellwidget.capture_dir_change)
+        self.shellwidget.sig_change_cwd.connect(self.update_cwd)

```

Comment:
Please remove this, it's not necessary because we can connect things directly to the shellwidget in the IPython console plugin.